homeassistant/components/lunatone/config_flow.py

```diff
@@ -0,0 +1,221 @@
+"""Config flow for Lunatone."""
+
+from __future__ import annotations
+
+import asyncio
+from enum import StrEnum
+import logging
+from typing import Any, Final
+
+import aiohttp
+from lunatone_rest_api_client import Auth, DALIScan, Info
+from lunatone_rest_api_client.models import StartScanData
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    SOURCE_RECONFIGURE,
+    SOURCE_USER,
+    ConfigFlow,
+    ConfigFlowResult,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.helpers import config_validation as cv, selector
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DALIDeviceScanMethod(StrEnum):
+    """DALI device scan methods."""
+
+    DO_NOTHING = "do_nothing"
+    CURRENT_DEVICE_LIST = "current_device_list"
+    SYSTEM_EXTENSION = "system_extension"
+    NEW_INSTALLATION = "new_installation"
+
+
+DEFAULT_DALI_DEVICE_SCAN_METHOD: Final = DALIDeviceScanMethod.DO_NOTHING
+DALI_DEVICE_SCAN_METHODS: Final[list[str]] = [
+    option.value for option in DALIDeviceScanMethod
+]
+
+DATA_SCHEMA: Final[vol.Schema] = vol.Schema(
+    {vol.Required(CONF_URL, default="http://"): cv.string},
+)
+RECONFIGURE_SCHEMA: Final[vol.Schema] = DATA_SCHEMA
+CONF_SCAN_METHOD: Final = "device_scan_method"
+DALI_SCAN_SCHEMA: Final[vol.Schema] = vol.Schema(
+    {
+        vol.Required(
+            CONF_SCAN_METHOD, default=DEFAULT_DALI_DEVICE_SCAN_METHOD
+        ): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=DALI_DEVICE_SCAN_METHODS,
+                translation_key=CONF_SCAN_METHOD,
+                mode=selector.SelectSelectorMode.LIST,
+            ),
+        ),
+    }
+)
+
+
+class LunatoneConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Lunatone config flow."""
+
+    VERSION = 0
+    MINOR_VERSION = 2
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.url: str | None = None
+        self.name: str | None = None
+        self.serial_number: int | None = None
+        self.dali_device_scan_task: asyncio.Task | None = None
```

Comment:
```suggestion
        self._url: str | None = None
        self._name: str | None = None
        self._serial_number: int | None = None
        self._dali_device_scan_task: asyncio.Task | None = None
```
can we prefix this since they are internal?